### PR TITLE
ETH-103: approve before transferToMemberInContract

### LINKED ISSUE #263: transferToMemberInContract requires approve
This probably should be done as part of transferToMemberInContract function within the library

Reported in Slack, ticket created on 2021-05-12

Report: 
```
The transferToMemberInContract function needs approval of the caller for transferring the tokens, however the streamr-client doesn't contains the approval.
```

### PR DESCRIPTION
**Pull request by jtakalai**
Fixes #263

### CODE REVIEW

File: src/dataunion/DataUnion.ts [hunk 1]
```diff
@@ -70,6 +70,20 @@ export type AmbMessageHash = string
 
 const log = debug('StreamrClient::DataUnion')
 
+const erc20AllowanceAbi = [{
```
Comment (teogeb) [inline src/dataunion/DataUnion.ts:73]: Move the ABI definition e.g. `to src/dataunion/abi.ts`?

**Review by timoxley — APPROVED** [review]
lgtm

File: src/dataunion/DataUnion.ts (hunk 1 above)
Comment (jtakalai) [reply src/dataunion/DataUnion.ts]: done

**Comment by jtakalai** [comment]
dataunions tests pass

**Pull request merged**